lib/ip_wrangler/iptables.rb

```diff
@@ -0,0 +1,231 @@
+module IpWrangler
+  class Iptables
+    $iptables_bin_path = '/usr/bin/sudo /sbin/iptables'
+    $awk_bin_path = '/usr/bin/awk'
+    $tail_bin_path = '/usr/bin/tail'
+    $grep_bin_path = '/bin/grep'
+
+    def initialize(chain_name, logger)
+      @chain_name = chain_name
+      @logger = logger
+    end
+
+    def rule_nat_port(public_ip, public_port, private_ip, private_port, protocol)
+      [Parameter.destination(public_ip),
+       Parameter.protocol(protocol),
+       Parameter.destination_port(public_port),
+       Parameter.jump('DNAT'),
+       Parameter.to_destination("#{private_ip}:#{private_port}")]
+    end
+
+    def rule_nat_ip(public_ip, private_ip)
+      rule_dnat = [Parameter.destination(public_ip),
+                   Parameter.jump('DNAT'),
+                   Parameter.to_destination(private_ip)]
+      rule_snat = [Parameter.source(private_ip),
+                   Parameter.jump('SNAT'),
+                   Parameter.to(public_ip)]
+
+      return rule_dnat, rule_snat
+    end
+
+    def append_nat_port(public_ip, public_port, private_ip, private_port, protocol)
+      rule = rule_nat_port public_ip, public_port, private_ip, private_port, protocol
+
+      execute Command.check_rule "#{@chain_name}_PRE", 'nat', rule
+      if $?.exitstatus == 1
+        execute Command.append_rule "#{@chain_name}_PRE", 'nat', rule
+      end
+    end
+
+    def append_nat_ip(public_ip, private_ip)
+      rule_dnat, rule_snat = rule_nat_ip public_ip, private_ip
+
+      execute Command.check_rule "#{@chain_name}_PRE", 'nat', rule_dnat
+      if $?.exitstatus == 1
+        execute Command.append_rule "#{@chain_name}_PRE", 'nat', rule_dnat
+      end
+      execute Command.check_rule "#{@chain_name}_POST", 'nat', rule_snat
+      if $?.exitstatus == 1
+        execute Command.append_rule "#{@chain_name}_POST", 'nat', rule_snat
+      end
+    end
+
+    def delete_nat_port(public_ip, public_port, private_ip, private_port, protocol)
+      rule = rule_nat_port public_ip, public_port, private_ip, private_port, protocol
+
+      command = Command.delete_rule_spec "#{@chain_name}_PRE", rule, 'nat'
+
+      execute command
+    end
+
+    def delete_nat_ip(public_ip, private_ip)
+      rule_dnat, rule_snat = rule_nat_ip public_ip, private_ip
+
+      command_dnat = Command.delete_rule_spec "#{@chain_name}_PRE", rule_dnat, 'nat'
+      command_snat = Command.delete_rule_spec "#{@chain_name}_POST", rule_snat, 'nat'
+
+      execute command_dnat, command_snat
+    end
+
+    def not_exists_nat_port?(public_ip, public_port, protocol, private_ip, private_port)
```

Comment:
Unused method argument - `private_ip`. If it's necessary, use `_` or `_private_ip` as an argument name to indicate that it won't be used.<br>Unused method argument - `private_port`. If it's necessary, use `_` or `_private_port` as an argument name to indicate that it won't be used.<br>Line is too long. [88/80]